### PR TITLE
[systemtest] Fixes for nightlies

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Tests fixes

### Description

After some PRs and new features it seems that some tests are failing.

-> naming of services was changed for Kafka and Zookeeper in `strimzi.io/name`
-> some listeners was switch: NODEPORT -> LOADBALANCER
-> removed the listeners from resource where was already set in the `defaultKafka` method in `KafkaResource`

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ListenersST,TracingST,CustomResourceStatusST

**Review by scholzj — APPROVED** [review]

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ListenersST,TracingST,CustomResourceStatusST

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 1]
```diff
@@ -430,21 +430,25 @@ void verifyLabelsForCRDs() {
...
-        List<String> servicesList = new ArrayList<>();
-        servicesList.add(clusterName + "-kafka-bootstrap");
-        servicesList.add(clusterName + "-kafka-brokers");
-        servicesList.add(clusterName + "-zookeeper-nodes");
-        servicesList.add(clusterName + "-zookeeper-client");
+        String kafkaServiceName = clusterName + "-kafka";
+        String zookeeperServiceName = clusterName + "-zookeeper";
 
-        for (String serviceName : servicesList) {
+        Map<String, String> servicesList = new HashMap<>();
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:436]: It's not servicesList but map )
```suggestion
        Map<String, String> servicesMap = new HashMap<>();
```

**Review by see-quick — APPROVED** [review]
Looks good :) but one nit.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ListenersST,TracingST,CustomResourceStatusST

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 2]
```diff
@@ -992,7 +992,7 @@ void testRegenerateCertExternalAddressChange() {
                     new GenericKafkaListenerBuilder()
                             .withName("external")
                             .withPort(9094)
-                            .withType(KafkaListenerType.NODEPORT)
+                            .withType(KafkaListenerType.LOADBALANCER)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:995]: Why this caused issues?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:995]: There was problem with certificate difference between `NODEPORT` and `LOADBALANCER` -> and also the original test had external listener `LOADBALANCER` as the tag shows. The difference in certificate was in several characters.

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:995]: The nodeport doesn't have any hostnames in cerificates so the certificates are same like without the nodeport. They are different for the loadbalancer.

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=CustomResourceStatusST

**Pull request merged**